src/platform_impl/linux/x11/util/geometry.rs

```diff
@@ -183,41 +183,71 @@ impl XConnection {
         window: ffi::Window,
         root: ffi::Window,
     ) -> Result<TranslatedCoords, XError> {
-        let mut translated_coords: TranslatedCoords = unsafe { mem::uninitialized() };
+        let mut x_rel_root = MaybeUninit::uninit();
+        let mut y_rel_root = MaybeUninit::uninit();
+        let mut child = MaybeUninit::uninit();
+
         unsafe {
             (self.xlib.XTranslateCoordinates)(
                 self.display,
                 window,
                 root,
                 0,
                 0,
-                &mut translated_coords.x_rel_root,
-                &mut translated_coords.y_rel_root,
-                &mut translated_coords.child,
+                x_rel_root.as_mut_ptr(),
+                y_rel_root.as_mut_ptr(),
+                child.as_mut_ptr(),
             );
         }
-        //println!("XTranslateCoordinates coords:{:?}", translated_coords);
-        self.check_errors().map(|_| translated_coords)
+
+        self.check_errors()?;
+
+        unsafe {
+            Ok(TranslatedCoords {
+                x_rel_root: x_rel_root.assume_init(),
+                y_rel_root: y_rel_root.assume_init(),
+                child: child.assume_init(),
+            })
+        }
     }
 
     // This is adequate for inner_size
     pub fn get_geometry(&self, window: ffi::Window) -> Result<Geometry, XError> {
-        let mut geometry: Geometry = unsafe { mem::uninitialized() };
+        let mut root = MaybeUninit::uninit();
```

Comment:
I think it should be safe to zero-initialize the whole struct, then pass pointers to the individual members in it.